api/app/Http/Controllers/StatsController.php

```diff
@@ -337,88 +344,59 @@ public function getDuoStats(string $period = '', string $sort = 'winlose'): Json
-            }
-            if ($stats['crawlscore'] !== 0 && $stats['totalgames'] !== 0) {
-                $stats['avgcrawlscore'] = $stats['crawlscore'] / $stats['totalgames'];
-            }
-
-            // set win/lose ratio
-            if ($stats['won'] !== 0 && $stats['lost'] !== 0) {
-                $stats['winlose'] = $stats['won'] / $stats['lost'];
+            if ($stats->won !== 0 && $stats->lost !== 0) {
+                $stats->winlose = round($stats->won / $stats->lost, 2);
             }
 
-            if ($stats['totalgames'] < 6) {
-                continue;
-            }
-
-            $stats['avgscore'] = round($stats['avgscore'], 2);
-            $stats['avgcrawlscore'] = round($stats['avgcrawlscore'], 2);
-            $stats['winlose'] = round($stats['winlose'], 2);
-
-            $data[] = $stats;
+            $data->push($stats);
         }
 
         switch ($sort) {
             case 'winlose':
-                usort($data, function ($a, $b) {
-                    return $a['winlose'] < $b['winlose'];
-                });
+                $data = $data->sortByDesc('winlose');
                 break;
             case 'win':
-                usort($data, function ($a, $b) {
-                    return $a['won'] < $b['won'];
-                });
+                $data = $data->sortByDesc('won');
                 break;
             case 'lose':
-                usort($data, function ($a, $b) {
-                    return $a['lost'] < $b['lost'];
-                });
+                $data = $data->sortByDesc('lost');
                 break;
             case 'avgscore':
-                usort($data, function ($a, $b) {
-                    return $a['avgscore'] < $b['avgscore'];
-                });
+                $data = $data->sortByDesc('avgscore');
                 break;
             case 'score':
-                usort($data, function ($a, $b) {
-                    return $a['totalscore'] < $b['totalscore'];
-                });
+                $data = $data->sortByDesc('totalscore');
                 break;
             case 'avgcrawl':
-                usort($data, function ($a, $b) {
-                    return $a['avgcrawlscore'] < $b['avgcrawlscore'];
-                });
+                $data = $data->sortByDesc('avgcrawlscore');
                 break;
             case 'crawl':
-                usort($data, function ($a, $b) {
-                    return $a['crawlscore'] < $b['crawlscore'];
-                });
+                $data = $data->sortByDesc('crawlscore');
                 break;
             case 'totalgames':
-                usort($data, function ($a, $b) {
-                    return $a['totalgames'] < $b['totalgames'];
-                });
+                $data = $data->sortByDesc('totalgames');
                 break;
         }
 
-        return new JsonResponse($data);
-    }
+        // Laravel doesn't return an array when using $data->toArray for some reason.
```

Comment:
A plain object doesn't implement the [Laravel interface Arrayable](https://github.com/laravel/framework/blob/5.0/src/Illuminate/Contracts/Support/Arrayable.php), so the raw value will be returned (see https://github.com/laravel/framework/blob/5.5/src/Illuminate/Support/Collection.php#L1608).

Reply:
Fixed with keyValue pipe